docs/specification/draft/basic/transports.mdx

```diff
@@ -287,6 +288,129 @@ protocol version 2024-11-05) as follows:
        the old HTTP+SSE transport, and should use that transport for all subsequent
        communication.
 
+## WebSocket
+
+This section defines bidirectional transport of MCP over a WebSocket connection which provides better support for maintaining long-lived connections particularly in cloud environments and through intermediaries. Long-lived connections are necessary for server-initiated notifications like `notifications/resources/updated` and `notifications/*/list_changed` which enable real-time use cases.
+
+The server **MUST** provide an HTTP endpoint path that supports WebSocket upgrade HTTP request. This **SHOULD** be the same endpoint as the Streamable HTTP transport.
+
+### Message Handling
+
+All messages **MUST** be sent as UTF-8 encoded text frames. Binary frames are not supported and **MUST** result in connection termination with close code `1003` (Unsupported Data).
```

Comment:
Is the purpose of this limitation to simplify the implementation? I'm not very strong about this, but at least in the Python SDK this wouldn't be an issue - we can handle the validation of both frame formats in ~4 lines of code.